### ISSUE TITLE
why the generated sentences are always same?

### ISSUE THREAD

**Issue by lknownothing** [lileipisces/NRT#1]
Is it because the hidden state does not influcence the generated word?

**Comment by lileipisces** [comment]
The real problem is that the model is not well learned. It could be caused by many factors, such as large L2 coefficient or small gradient clipping value, which would limit the model parameters' learning, and thus influence the results.

I reported the L2 issue in my paper [PETER](https://lileipisces.github.io/files/ACL21-PETER-paper.pdf). Please check section 5.3. During the implementation of Transformer, I also encountered this problem, see my discussion on [PyTorch examples](https://github.com/pytorch/examples/issues/846).
